web/src/engine/websites/Zebrack.ts

```diff
@@ -0,0 +1,347 @@
+            const gravureId = uri.pathname.match(/\/gravure\/(\d+)$/)[1];
+            const data = await this.fetchGravureDetail(gravureId);
+            return new Manga(this, provider, uri.pathname, data.gravure.name.trim());
+        }
+
+        const titleId = uri.pathname.match(/\/title\/(\d+)/)[1];
+        const data = await this.fetchTitleDetail(titleId);
+        return new Manga(this, provider, uri.pathname, data.titleDetailView.titleName.trim());
+
+    }
+    async fetchMagazineDetail(magazineId: string, magazineIssueId: string): Promise<MagazineDetailViewV3> {
+        const uri = new URL('/api/v3/magazine_issue_detail', this.apiURL);
+        uri.searchParams.set('os', 'browser');
+        uri.searchParams.set('magazine_id', magazineId);
+        uri.searchParams.set('magazine_issue_id', magazineIssueId);
+        const request = new FetchRequest(uri.href);
+        return FetchProto<MagazineDetailViewV3>(request, protoTypes, 'Zebrack.MagazineDetailViewV3');
+    }
+
+    async fetchGravureDetail(gravureId: string): Promise<GravureDetailViewV3> {
+        const uri = new URL('/api/v3/gravure_detail', this.apiURL);
+        uri.searchParams.set('os', 'browser');
+        uri.searchParams.set('gravure_id', gravureId);
+        const request = new FetchRequest(uri.href);
+        return FetchProto<GravureDetailViewV3>(request, protoTypes, 'Zebrack.GravureDetailViewV3');
+    }
+
+    async fetchTitleDetail(titleId: string): Promise<ZebrackResponse> {
+        const uri = new URL('/api/browser/title_detail', this.apiURL);
+        uri.searchParams.set('os', 'browser');
+        uri.searchParams.set('title_id', titleId);
+        const request = new FetchRequest(uri.href);
+        return FetchProto<ZebrackResponse>(request, protoTypes, this.responseRootType);
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const parts = manga.Identifier.split('/');
+        let type = parts[3] || 'chapter';
+        if (['magazine', 'gravure'].includes(parts[1])) {
+            type = parts[1];
+        }
+        if (type === 'chapter') {
+            const id = parts[2];
+            const data = await this.fetchChapterList(id);
+            const chapters: ChapterV3[] = [];
+            data.groups.forEach(group => {
+                chapters.push(...group.chapters);
+            });
+            return chapters.map(chapter => new Chapter(this, manga, `chapter/${chapter.titleId}/${chapter.id}`, chapter.mainName));
+        }
+
+        if (type === 'gravure') {
+            return [new Chapter(this, manga, manga.Identifier.slice(1), manga.Title)];
+        }
+
+        if (type === 'magazine') {
+            const magazineId = parts[2];
+            const magazineIssueId = parts[4];
+            return [new Chapter(this, manga, `magazine/${magazineId}/${magazineIssueId}`, manga.Title)];
+        }
+
+        if (type === 'volume_list' || type === 'volume') {
```

Comment:
The`volume_list` key is missing in `ValidateMangaURL` check, is this by intention?